File: packages/sdui-common/src/main/java/nl/q42/common/components/TextColor.java
package nl.q42.common.components;

public enum TextColor
{
  PRIMARY,
  SECONDARY,
  TERTIARY,
  DISABLED,
  ERROR,
  SUCCESS,
  INFO,
  CUSTOM;

  @Override
  public String toString()
  {
    return String.format("%d", this.ordinal());
  }

  public static TextColor from(int value)
  {
    var values = TextColor.values();
    var withinRange = value >= 0 && value < values.length;

    return !withinRange
           ? PRIMARY : values[value];
  }
}

File: packages/sdui-common/src/main/kotlin/nl/q42/common/components/TextColor.kt
package nl.q42.common.components

enum class TextColor {
    PRIMARY,
    SECONDARY,
    TERTIARY,
    DISABLED,
    ERROR,
    SUCCESS,
    INFO,
    CUSTOM;

    override fun toString(): String {
        return String.format("%d", this.ordinal)
    }

    companion object {
        fun from(value: Int): TextColor? {
            val values: Array<TextColor?> = TextColor.entries.toTypedArray()
            val withinRange = value >= 0 && value < values.size

            return if (!withinRange) TextColor.PRIMARY else
                values[value]
        }
    }
}
